godel_surface_detection/src/scan/profilimeter_scan.cpp

```diff
@@ -0,0 +1,244 @@
+    double max_x = boundary[0].x;
+    double max_y = boundary[0].y;
+
+    for (std::size_t i = 1; i < boundary.size(); ++i)
+    {
+      const PolygonPt& pt = boundary[i];
+      min_x = std::min(pt.x, min_x);
+      min_y = std::min(pt.y, min_y);
+      max_x = std::max(pt.x, max_x);
+      max_y = std::max(pt.y, max_y); 
+    }
+
+    RotatedRect result;
+    result.a = 0.0; // non-rotated, hence the simple bounding box
+    result.x = (max_x + min_x) / 2;
+    result.y = (max_y + min_y) / 2;
+    result.w = max_x - min_x;
+    result.h = max_y - min_y;
+    return result;
+  }
+
+  // Vertically or horizontally?
+  std::vector<RotatedRect> sliceBoundingBox(const RotatedRect& bbox, double slice_width, double overlap)
+  {
+    // we assume slice_width > overlap
+    int n_slices;
+    float dx, dy; // unit vector
+    float x, y; // start position
+    float w, h;
+
+    float adjusted_width = slice_width - overlap;
+
+    // Determine which way the slice should go
+    if (bbox.w < bbox.h)
+    {
+      // walk horizontally, vertical slices
+      n_slices = static_cast<int>(bbox.w / (adjusted_width)) + 1;
+      
+      dx = std::cos(bbox.a);
+      dy = std::sin(bbox.a);
+
+      x = bbox.x - dx * (bbox.w - adjusted_width) / 2.0;
+      y = bbox.y - dy * (bbox.w - adjusted_width) / 2.0;
+
+      w = adjusted_width;
+      h = bbox.h;
+    }
+    else
+    {
+      // walk vertically, horizontal slices
+      n_slices = static_cast<int>(bbox.h / (adjusted_width)) + 1;
+      
+      dx = -std::sin(bbox.a);
+      dy = std::cos(bbox.a);
+
+      x = bbox.x - dx * (bbox.h - adjusted_width) / 2.0;
+      y = bbox.y - dy * (bbox.h - adjusted_width) / 2.0; 
+
+      w = bbox.w;
+      h = adjusted_width;
+    }
+    
+    std::vector<RotatedRect> slices;
+    slices.reserve(n_slices);
+
+    dx *= adjusted_width;
+    dy *= adjusted_width;
+
+    for (int i = 0; i < n_slices; ++i)
+    {
+      RotatedRect temp;
+      temp.a = bbox.a;
+      temp.x = x;
+      temp.y = y;
+      temp.w = w;
+      temp.h = h;
+      slices.push_back(temp);
+      x += dx;
+      y += dy;
+    }
+
+    return slices;
+  }
+
+  std::vector<PolygonPt> interpolateAlongAxis(const RotatedRect& rect, double ds)
+  {
+    std::vector<PolygonPt> pts;
+    double dx, dy; // 
+    double x, y;
+    int n;
+    if (rect.w > rect.h)
+    {
+      n = rect.w / ds;
+      dx = std::cos(rect.a);
+      dy = std::sin(rect.a);
+      x = rect.x - dx * rect.w/2.0;
+      y = rect.y - dy * rect.w/2.0; 
+    }
+    else
+    {
+      n = rect.h / ds;
+      dx = -std::sin(rect.a);
+      dy = std::cos(rect.a);
+      x = rect.x - dx * rect.h/2.0;
+      y = rect.y - dy * rect.h/2.0;
+    }
+
+    // pts.reserve(n);
+
+    // transform unit vector
+    dx *= ds; 
+    dy *= ds;
+
+    for (int i = 0; i < n; ++i)
+    {
+      PolygonPt pt;
+      pt.x = x;
+      pt.y = y;
+      x += dx;
+      y += dy;
+      pts.push_back(pt);
+    }
+
+    return pts;
```

Comment:
Could this function return really long vectors?  If so then it may result into rather expensive copy operations.  Base on the CMakeList and dependency list it doesn't look like move semantics is enabled for this package, but I could be wrong